classes/Tools.php

```diff
@@ -725,43 +730,102 @@ public static function getCldr(Context $context = null, $language_code = null)
     }
 
     /**
-     * Return price with currency sign for a given product.
+     * Return price with currency sign for a given product
      *
-     * @param float $price Product price
-     * @param object|array $currency Current currency (object, id_currency, NULL => context currency)
+     * @deprecated Since 1.7.5.0. Please use Locale::formatPrice() instead
+     * @see PrestaShop\PrestaShop\Core\Localization\Locale
+     *
+     * @param float $price
+     *  Product price
+     *
+     * @param object|array $currency
+     *  Current currency (object, id_currency, NULL => context currency)
+     *
+     * @param bool $no_utf8
+     *  Not used anymore
      *
-     * @return string Price correctly formated (sign, decimal separator...)
-     *                if you modify this function, don't forget to modify the Javascript function formatCurrency (in tools.js)
+     * @param Context|null $context
+     *
+     * @return string Price correctly formatted (sign, decimal separator...)
+     * if you modify this function, don't forget to modify the Javascript function formatCurrency (in tools.js)
+     *
+     * @throws LocalizationException
      */
     public static function displayPrice($price, $currency = null, $no_utf8 = false, Context $context = null)
     {
+        @trigger_error(
+            'Tools::displayPrice() is deprecated since version 1.7.5.0. '
+            . 'Use PrestaShop\PrestaShop\Core\Localization\Locale::formatPrice() instead.',
+            E_USER_DEPRECATED
+        );
+
         if (!is_numeric($price)) {
             return $price;
         }
-        if (!$context) {
-            $context = Context::getContext();
-        }
-        if ($currency === null) {
-            $currency = $context->currency;
-        } elseif (is_int($currency)) {
-            $currency = Currency::getCurrencyInstance((int) $currency);
+
+        $context  = $context ?: Context::getContext();
+        $currency = $currency ?: $context->currency;
+
+        if (is_int($currency)) {
```

Comment:
```php
if (is_int($currency)) {
  $currency = Currency::getCurrencyInstance($currency);
} else {
  $currency = $context->currency;
}
```

Reply:
since property $currency is public, we cannot be sure it is always an object... :(